browser/sync/BUILD.gn

```diff
@@ -1,4 +1,6 @@
 source_set("sync") {
+  check_includes = false
```

Comment:
nit: probably better to add a comment with a tracking issue to remove this in the future.

Reply:
https://github.com/brave/brave-browser/issues/12977